internal/restorer/restorer.go

```diff
@@ -285,58 +282,107 @@ func (res *Restorer) Snapshot() *restic.Snapshot {
 	return res.sn
 }
 
-// VerifyFiles reads all snapshot files and verifies their contents
+// Number of workers in VerifyFiles.
+const nVerifyWorkers = 8
+
+// VerifyFiles checks whether all regular files in the snapshot res.sn
+// have been successfully written to dst. It stops when it encounters an
+// error. It returns that error and the number of files it has successfully
+// verified.
 func (res *Restorer) VerifyFiles(ctx context.Context, dst string) (int, error) {
-	// TODO multithreaded?
+	type mustCheck struct {
+		node *restic.Node
+		path string
+	}
 
-	count := 0
-	err := res.traverseTree(ctx, dst, string(filepath.Separator), *res.sn.Tree, treeVisitor{
-		enterDir: func(node *restic.Node, target, location string) error { return nil },
-		visitNode: func(node *restic.Node, target, location string) error {
-			if node.Type != "file" {
-				return nil
-			}
+	var (
+		nchecked uint64
+		work     = make(chan mustCheck, 2*nVerifyWorkers)
```

Comment:
What is the reason for `2*nVerifyWorkers`? To have a queue with tasks to process? I'm especially wondering about why `2*nVerifyWorkers` an not `nVerifyWorkers` or `3*nVerifyWorkers`? (`2*nVerifyWorkers` seems reasonable though.)